src/lib/location.ts

```diff
@@ -107,80 +108,87 @@ export async function resolveLocation(
+
+      const state = await db.get<DbGeography>(
+        `SELECT * FROM geographies
+        WHERE type = 'state' AND state = ?`,
+        result.address_components.state,
+      );
+      if (state) {
+        geographies.push({ ...state, intersection_proportion: 1.0 });
+      }
+
+      const county = await db.get<DbGeography>(
+        `SELECT * FROM geographies
+        WHERE type = 'county' AND county_fips = ?`,
+        censusInfo.county_fips,
+      );
+      if (county) {
+        geographies.push({ ...county, intersection_proportion: 1.0 });
+      }
+
+      const point = turf.point([result.location.lng, result.location.lat]);
+
+      // There is a relatively small number of 'custom' geographies, so running
+      // the "contains" computation over all of them is quick. If necessary, we
+      // can speed this up by using a spatial index and/or caching the parsed
+      // GeoJSON in memory.
+      (
+        await db.all<DbGeography[]>(
+          `SELECT * FROM geographies WHERE type = 'custom'`,
+        )
       )
-    )
-      .filter(geo => {
-        const geometry = JSON.parse(geo.geometry!);
-        let contains = false;
-
-        // Check whether the point is inside any sub-geometry of the overall
-        // geometry object (which is usually a FeatureCollection).
-        turf.geomEach(geometry, subgeometry => {
-          if (turf.booleanPointInPolygon(point, subgeometry)) {
-            contains = true;
-            // Returning false from the geomEach callback terminates the
-            // geomEach loop early.
-            return false;
-          }
-        });
-        return contains;
-      })
-      .forEach(geo =>
-        geographies.push({ ...geo, intersection_proportion: 1.0 }),
+        .filter(geo => {
+          const geometry = JSON.parse(geo.geometry!);
+          let contains = false;
+
+          // Check whether the point is inside any sub-geometry of the overall
+          // geometry object (which is usually a FeatureCollection).
+          turf.geomEach(geometry, subgeometry => {
+            if (turf.booleanPointInPolygon(point, subgeometry)) {
+              contains = true;
+              // Returning false from the geomEach callback terminates the
+              // geomEach loop early.
+              return false;
+            }
+          });
+          return contains;
+        })
+        .forEach(geo =>
+          geographies.push({ ...geo, intersection_proportion: 1.0 }),
+        );
+
+      return {
+        state: result.address_components.state,
+        zcta:
+          (await getZctaFromZip(db, result.address_components.zip)) ??
+          result.address_components.zip,
+        geographies,
+        tract_geoid: GEOCODIO_LOW_PRECISION.has(result.accuracy_type)
+          ? undefined
+          : censusInfo.county_fips + censusInfo.tract_code,
+      };
+    } catch (error) {
+      throw new InvalidInputError(
+        `Unable to geocode address: ${address}`,
```

Comment:
So I was kinda wrong when I explained the plan before. The problem is that `resolveLocation` is called from the top level code in `v1.ts` outside of any try blocks. Its (implicit) contract is that it never throws; it indicates error by returning null. (This is why the visible text in the screenshot still says "internal server error"; it's still serving a 500.)

If you just return null here, the right thing should happen: v1.ts will return a localized error message, with a 404 status, that says we can't locate that address.